MagneticField/VolumeBasedEngine/src/MagGeometry.cc

```diff
@@ -49,25 +46,49 @@ MagGeometry::MagGeometry(int geomVersion,
   vector<double> rBorders;
 
   for (vector<MagBLayer const*>::const_iterator ilay = theBLayers.begin(); ilay != theBLayers.end(); ++ilay) {
-    if (verbose::debugOut)
-      cout << "  Barrel layer at " << (*ilay)->minR() << endl;
+    LogTrace("MagGeometry") << "  Barrel layer at " << (*ilay)->minR() << endl;
     //FIXME assume layers are already sorted in minR
-    rBorders.push_back((*ilay)->minR());
+    rBorders.push_back((*ilay)->minR() * (*ilay)->minR());
   }
 
   theBarrelBinFinder = new MagBinFinders::GeneralBinFinderInR<double>(rBorders);
 
-  if (verbose::debugOut) {
-    for (vector<MagESector const*>::const_iterator isec = theESectors.begin(); isec != theESectors.end(); ++isec) {
-      cout << "  Endcap sector at " << (*isec)->minPhi() << endl;
-    }
+#ifdef EDM_ML_DEBUG
+  for (vector<MagESector const*>::const_iterator isec = theESectors.begin(); isec != theESectors.end(); ++isec) {
+    LogTrace("MagGeometry") << "  Endcap sector at " << (*isec)->minPhi() << endl;
   }
+#endif
 
   //FIXME assume sectors are already sorted in phi
   //FIXME: PeriodicBinFinderInPhi gets *center* of first bin
   int nEBins = theESectors.size();
   if (nEBins > 0)
     theEndcapBinFinder = new PeriodicBinFinderInPhi<float>(theESectors.front()->minPhi() + Geom::pi() / nEBins, nEBins);
+
+  // Compute barrel dimensions based on geometry version
+  switch (geomVersion >= 120812 ? 0 : (geomVersion >= 90812 ? 1 : 2)) {
+    case 0:  // since 120812
+      barrelRsq1 = 172.400f * 172.400f;
+      barrelRsq2 = 308.735f * 308.735f;
+      barrelZ0 = 350.000f;
+      barrelZ1 = 633.290f;
+      barrelZ2 = 662.010f;
+      break;
+    case 1:  // version 90812 (no longer in use)
+      barrelRsq1 = 172.400f * 172.400f;
+      barrelRsq2 = 308.755f * 308.755f;
+      barrelZ0 = 350.000f;
+      barrelZ1 = 633.890f;
+      barrelZ2 = 662.010f;
+      break;
+    case 2:  // versions 71212, 90322
+      barrelRsq1 = 172.400f * 172.400f;
+      barrelRsq2 = 308.755f * 308.755f;
+      barrelZ0 = 350.000f;
+      barrelZ1 = 633.290f;
+      barrelZ2 = 661.010f;
+      break;
```

Comment:
this is apparently relocating some magic numbers from what was earlier in ```MagGeometry::inBarrel```.
There we had ```// FIXME: Get these dimensions from the builder.```
Is this available from the builder?
Minimally, it's still worth to keep the FIXME comment.

Reply:
I put the comment back,
I wrote the FIXME a dozen years ago in the hope I would have figured out a reasonable way to detect the dimensions automatically or to label volumes appropriately in the geometry. I haven't since. A sufficiently generic solution would require adding specifications to the geometry and some logic, which I don't think is worth given that these numbers are not expected to change (last change was 8 years ago and none is in sight in the future)